src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorParserOptions.cs

```diff
@@ -191,4 +192,15 @@ public RazorParserOptions WithFlags(
             ? this
             : new(LanguageVersion, FileKind, Directives, CSharpParseOptions, flags);
     }
+
+    public bool Equals(RazorParserOptions? other)
```

Comment:
Please override `Equals` and `GetHashCode` so that value equality is always used for `RazorParserOptions` and not just some of the time. Or, make `RazorParserOptions` a record.